### PR TITLE
Use Algolia plugin

### PR DESCRIPTION
**Pull request by marvinchin**
Previously, Algolia was integrated manually. This PR switches to use a plugin to support Algolia search.

Requires https://github.com/MarkBind/markbind/pull/718

### CODE REVIEW

File: _markbind/plugins/algolia.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+module.exports = {
+  getLinks: (content, pluginContext, frontMatter, utils) => [utils.buildStylesheet(ALGOLIA_CSS_URL)],
+  getScripts: (content, pluginContext, frontMatter, utils) =>
+    [
+      utils.buildScript(ALGOLIA_JS_URL),
+      buildAlgoliaInitScript(pluginContext.apiKey, pluginContext.indexName),
+    ],
+};
```
Comment (damithc) [inline _markbind/plugins/algolia.js]: umm... do we have to add this file manually? Looks like a standard file MarkBind can provide by default.

Comment (marvinchin) [reply _markbind/plugins/algolia.js]: Oops, my bad - didn't realise that MarkBind can use plugins in it's own `plugins` directory as well. 😅

**Comment by damithc** [comment]
This PR will not work until MarkBind's algolia integration feature is released to the public, am I right?

**Comment by marvinchin** [comment]
Yes, that's correct! I will add WIP to the title until it is ready to be merged. Sorry if there was any confusion!

**Comment by damithc** [comment]
> Yes, that's correct! I will add WIP to the title until it is ready to be merged. Sorry if there was any confusion!

No worries. I jumped the gun :-p

**Comment by marvinchin** [comment]
The Algolia plugin has been released 🙂

File: _markbind/layouts/default/navigation.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <navbar placement="top" type="inverse">
   <a slot="brand" href="/" title="Home" class="navbar-brand">Software Engineering for Self-Directed Learners</a>
   <li slot="right">
-    <input id="searchbar" placeholder="Search" class="form-control"> 
+    <searchbar placeholder="Search" algolia menu-align-right>
```
Comment (damithc) [inline _markbind/layouts/default/navigation.md]: Shouldn't there be a `/>`?

Comment (marvinchin) [reply _markbind/layouts/default/navigation.md]: Fixed 🙂 

Preview was generated using an older version of MarkBind. Latest preview should be working now!

Comment (damithc) [reply _markbind/layouts/default/navigation.md]: > Fixed

Thanks. I noticed the same error in the MarkBind user guide.

Comment (marvinchin) [reply _markbind/layouts/default/navigation.md]: Made a PR to fix that as well. Thanks for catching it!

**Pull request merged**